packages/sdk/test/e2e/node/apiClient.spec.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { BearerTokenAuthProvider, createApiClient } from "../../../src";
+import * as http from "http";
+
+describe("ApiClient Tests - Node", () => {
+  const host = "localhost";
+  const port = 53001;
+  const apiBaseUrl = `http://${host}:${port}`;
+  const server = http.createServer((req, res) => {
+    res.writeHead(200);
+    res.end("Successfully create http server for ApiClient E2E test.");
+  });
+
+  before(() => {
+    server.listen(port, host, () => {
+      console.log(`Server is running on http://${host}:${port}`);
+    });
+  });
+
+  after(() => {
+    server.close(() => {
+      console.log(`Server closed`);
+    });
+  });
+
+  it("can connect to existing API using bearer token auth provider", async function () {
+    // Arrange
+    const bearerTokenAuthProvider = new BearerTokenAuthProvider(async () => "test-bearer-token");
+    const apiClient = createApiClient(apiBaseUrl, bearerTokenAuthProvider);
+
+    // Act
+    const res = await apiClient.get("/foo");
+
+    // Assert
+    assert.equal(res.config.baseURL, apiBaseUrl);
+    assert.equal(res.config.url, "/foo");
+    assert.equal(res.config.headers!["Authorization"], "Bearer test-bearer-token");
```

Comment:
I saw the test server does not set response headers, why we can get the response headers here?

Reply:
by the way, can we rename the file to bearerTokenAuthProvider.spec.ts? Since we mainly tests the auth provider, not createApiClient function here.

Reply:
I was referring to the `res.config.headers` which according to [axios API: response schema](https://axios-http.com/docs/res_schema) ,  `config` is the config that was provided to `axios` for the request.

Reply:
I was thinking having several `it()` function to test different auth provider e2e test in this same test file. How do you think?

Reply:
Fiexed by verifying authentication info with request header from server side.